docs/guides/tutorials/chatgpt_bot.rst

```diff
@@ -0,0 +1,2305 @@
+  ``dbschema`` directory in the project root.
+- Confirm the instance is running with the ``edgedb instance list`` command.
+  Search for the name of the instance you've just created (``docs_chatbot`` if
+  you're following along) and check the status. (Don't worry if the status is
+  "inactive"; the status will change to "running" automatically when you
+  connect to the instance.)
+- Confirm you can connect to the created instance by running ``edgedb`` in the
+  terminal to connect to it via REPL or by running ``edgedb ui`` to connect
+  using the UI.
+
+
+Configure the environment
+-------------------------
+
+Create a ``.env.local`` file in the root of your new Next.js project.
+
+.. code-block:: bash
+
+    $ touch .env.local
+
+We're going to add a couple of variables to that file to configure EdgeDB.
+We'll need to run a command on our new instance to get the value for one of
+those. Since the runtime we'll be using in our Next.js project does not have
+file system access, we need to provide a DSN to tell the project how to
+connect. To get that, run this command:
+
+.. code-block:: bash
+
+    $ edgedb instance credentials --insecure-dsn
+
+Copy what it logs out. Open the ``.env.local`` file in your text editor and add
+this to it:
+
+.. code-block:: typescript
+
+    EDGEDB_DSN=<your-dsn>
+    EDGEDB_CLIENT_TLS_SECURITY="insecure"
+
+Replace ``<your-dsn>`` with the value you copied earlier.
+
+We're going to be using the EdgeDB HTTP client a bit later to connect to our
+database, but it requires a trusted TLS/SSL certificate. Local development
+instances use self signed certificates, and using HTTPS with these certificates
+will result in an error. To work around this error, we allow the client to
+ignore TLS by setting the ``EDGEDB_CLIENT_TLS_SECURITY`` variable to
+``"insecure"``. Bear in mind that this is only for local development, and you
+should always use TLS in production.
+
+We need to set one more environment variable, but first we have to get an API
+key.
+
+
+Prepare the OpenAI API client
+-----------------------------
+
+We need an API key from OpenAI in order to make the calls we need to make this
+app work. To get one:
+
+1. Log in or sign up to the `OpenAI platform
+   <https://platform.openai.com/account/api-keys>`_.
+2. Create new `secret key <https://platform.openai.com/account/api-keys>`_.
+
+.. warning::
+
+    Don't forget: you may need to start a paid account if you do not have any
+    API free trial credits remaining.
+
+Copy the new key. Re-open your ``.env.local`` file and add it like this:
+
+.. code-block:: -diff
+
+      EDGEDB_DSN=<your-dsn>
+      EDGEDB_CLIENT_TLS_SECURITY="insecure"
+    + OPENAI_API_KEY="<your-openai-api-key>"
+
+Instead of ``<your-openai-api-key>``, paste in the key you just created.
+
+While we're here, let's get that key ready to be used. We will be making calls
+to the OpenAI API. We'll create a ``utils`` module and export a function from
```

Comment:
Maybe we can write something like:
`We’ll create a `utils` module and define in it a `initOpenAIClient` function that initialises an OpenAI client. We will reuse this function in every file we need a new client instance.`

Reply:
Right. Because we're not actually reusing a single client. Makes sense.